src/app/server/Server.cpp

```diff
@@ -176,10 +175,7 @@ CHIP_ERROR Server::Init(const ServerInitParams & initParams)
     // 1) Provider initialization (under SetDataModelProvider) happens after
     //    SetSafeAttributePersistenceProvider, since the provider can then use
     //    the safe persistence provider to implement and initialize its own attribute persistence logic.
-    // 2) For now, provider initialization happens before InitDataModelHandler(), which depends
-    //    on atttribute persistence being already set up before it runs.  Longer-term, the logic from
-    //    InitDataModelHandler should just move into the codegen provider.
-    chip::app::InteractionModelEngine::GetInstance()->SetDataModelProvider(initParams.dataModelProvider);
+    app::InteractionModelEngine::GetInstance()->SetDataModelProvider(initParams.dataModelProvider);
```

Comment:
OK, so this is moving the InitDataModelHandler() call to a _lot_ earlier in the startup sequence, no?

In particular, this will not initialize clusters before the fabric table is initialized, etc, etc.

This does not look right to me.

Reply:
Maybe after https://github.com/project-chip/connectedhomeip/pull/36658 we can place the set data model handler at the end ... it seems difficult to track which ordering was intentional or not

Reply:
Move it back to the original place